prefix.go

```diff
@@ -140,10 +140,18 @@ type Usage struct {
 }
 
 func (i *ipamer) NewPrefix(cidr string) (*Prefix, error) {
+	existingPrefixes, err := i.storage.ReadAllPrefixCidrs()
+	if err != nil {
+		return nil, err
+	}
 	p, err := i.newPrefix(cidr, "")
 	if err != nil {
 		return nil, err
 	}
+	err = i.PrefixesOverlapping(existingPrefixes, []string{p.Cidr})
```

Comment:
Without lock over all statements (getAllPrefixes, checkOverlap and createPrefix) this is not threadsafe and duplicates can still occur.

Reply:
Done, still thinking if we should, albeit breaks API, remove the ipamer.PrefixFrom Signature and do all work in ipam.NewPrefix.